uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -252,6 +252,28 @@ impl From<&&Type> for FfiType {
     }
 }
 
+// A trait so various things can turn into a type.
+pub trait AsType: core::fmt::Debug {
+    fn as_type(&self) -> Type;
+}
+
+impl AsType for Type {
+    fn as_type(&self) -> Type {
+        self.clone()
```

Comment:
This is a drawback to this approach, but a really minor one.  I think one of the reasons for the initial `CodeDispatch` system was that I was too afraid of the cloning.